In [2]:
import pandas as pd
import numpy as np

df = pd.read_excel("Fabric data.xlsx")
df

Fabric_length
0           151.2
1           160.3
2           147.5
3           149.2
4           159.2
5           155.6
6           148.1
7           154.8
8           156.8
9           167.5
10          164.5
11          147.9
12          154.1
13          159.4
14          147.5
15          148.2
16          154.2
17          148.1
18          163.1
19          155.2
20          154.6
21          155.4
22          158.2
23          157.6
24          158.4

- Given : n = 25, μ = 150cm, σ = 5.5cm, α = 0.05

#### Step-1 : Formulate the Hypothesis
    - H0 : μ == 150cm
    - H1 : μ != 150cm
        

#### Step-2 : Check alpha value 
i.e. α = 0.05
    

#### Step-3 : Check for Normality


In [4]:
df["Fabric_length"].skew() 

0.29650632012590666

The skewness is between -1 to 1. So, the data is normally distributed.

#### Step-4 : Select the test based on the problem statement

- It is a 1 Sample Z-test with both tailed test because we have identify population mean is equal or not and standard deviation is given.

#### Step-5 : Calculate the P-value(Probability value) using z-test formulae.

In [7]:
from statsmodels.stats.weightstats import ztest
ztest = ztest(df["Fabric_length"],value=150,alternative="two-sided",ddof=0)
ztest

(4.581553935379281, 4.615335331740764e-06)

#### or 

#### ztest = (x̄ - μ)/(σ/√n)

In [11]:
from math import sqrt

ztest = (df["Fabric_length"].mean()-150)/(5.5/sqrt(25))
ztest                                   

4.603636363636356

#### Step-6 : Check Based on calculated p-value, either accept or reject the null hypothesis.

- As the p-value is less then the alpha ( 4.60% < 5% or 0.00000461 < 0.05), so p-value lies in rejection area
- So, reject the Null Hypothesis and accept Alternative Hypothesis, means μ != 150cm so action is required.